base/dict.jl

```diff
@@ -189,6 +189,13 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(dictionary)
```

Comment:
I think that's a bug in the documentation, probably from a time when `similar` wasn't defined for other types. It would make sense to replace `array` with `a::AbstractArray`, since that docstring cannot say anything about what will happen with other types.